Can not build with cmake when enable prometheus exporter

**Describe your environment**

Linux

**Steps to reproduce**

Enable legacy prometheus exporter and prebuilt prometheus-cpp.

**What is the expected behavior?**
Build success

**What is the actual behavior?**

```bash
Building with standard library types...
CMake Error at exporters/prometheus/CMakeLists.txt:34 (install):
  install TARGETS given target "pull" which does not exist.


-- Configuring incomplete, errors occurred!
```

--- a/cmake/opentelemetry-cpp-config.cmake.in
+++ b/cmake/opentelemetry-cpp-config.cmake.in
@@ -20,30 +20,30 @@
 #   OPENTELEMETRY_VERSION           - Version of opentelemetry-cpp.
 #
 # ::
-#   opentelemetry-cpp::api                        - Imported target of opentelemetry-cpp::api
-#   opentelemetry-cpp::sdk                        - Imported target of opentelemetry-cpp::sdk
-#   opentelemetry-cpp::ext                        - Imported target of opentelemetry-cpp::ext
-#   opentelemetry-cpp::version                    - Imported target of opentelemetry-cpp::version
-#   opentelemetry-cpp::common                     - Imported target of opentelemetry-cpp::common
-#   opentelemetry-cpp::trace                      - Imported target of opentelemetry-cpp::trace
-#   opentelemetry-cpp::metrics                    - Imported target of opentelemetry-cpp::metrics
-#   opentelemetry-cpp::logs                       - Imported target of opentelemetry-cpp::logs
-#   opentelemetry-cpp::in_memory_span_exporter    - Imported target of opentelemetry-cpp::in_memory_span_exporter
-#   opentelemetry-cpp::otlp_recordable            - Imported target of opentelemetry-cpp::otlp_recordable
-#   opentelemetry-cpp::otlp_grpc_exporter         - Imported target of opentelemetry-cpp::otlp_grpc_exporter
-#   opentelemetry-cpp::otlp_grpc_log_exporter     - Imported target of opentelemetry-cpp::otlp_grpc_log_exporter
-#   opentelemetry-cpp::otlp_http_client           - Imported target of opentelemetry-cpp::otlp_http_client
-#   opentelemetry-cpp::otlp_http_exporter         - Imported target of opentelemetry-cpp::otlp_http_exporter
-#   opentelemetry-cpp::otlp_http_log_exporter     - Imported target of opentelemetry-cpp::otlp_http_log_exporter
-#   opentelemetry-cpp::ostream_log_exporter       - Imported target of opentelemetry-cpp::ostream_log_exporter
-#   opentelemetry-cpp::ostream_metrics_exporter   - Imported target of opentelemetry-cpp::ostream_metrics_exporter
-#   opentelemetry-cpp::ostream_span_exporter      - Imported target of opentelemetry-cpp::ostream_span_exporter
-#   opentelemetry-cpp::prometheus_exporter        - Imported target of opentelemetry-cpp::prometheus_exporter
-#   opentelemetry-cpp::elasticsearch_log_exporter - Imported target of opentelemetry-cpp::elasticsearch_log_exporter
-#   opentelemetry-cpp::etw_exporter               - Imported target of opentelemetry-cpp::etw_exporter
-#   opentelemetry-cpp::jaeger_trace_exporter      - Imported target of opentelemetry-cpp::jaeger_trace_exporter
-#   opentelemetry-cpp::zpages                     - Imported target of opentelemetry-cpp::zpages
-#   opentelemetry-cpp::http_client_curl           - Imported target of opentelemetry-cpp::http_client_curl
+#   opentelemetry-cpp::api                            - Imported target of opentelemetry-cpp::api
+#   opentelemetry-cpp::sdk                            - Imported target of opentelemetry-cpp::sdk
+#   opentelemetry-cpp::ext                            - Imported target of opentelemetry-cpp::ext
+#   opentelemetry-cpp::version                        - Imported target of opentelemetry-cpp::version
+#   opentelemetry-cpp::common                         - Imported target of opentelemetry-cpp::common
+#   opentelemetry-cpp::trace                          - Imported target of opentelemetry-cpp::trace
+#   opentelemetry-cpp::metrics                        - Imported target of opentelemetry-cpp::metrics
+#   opentelemetry-cpp::logs                           - Imported target of opentelemetry-cpp::logs
+#   opentelemetry-cpp::in_memory_span_exporter        - Imported target of opentelemetry-cpp::in_memory_span_exporter
+#   opentelemetry-cpp::otlp_recordable                - Imported target of opentelemetry-cpp::otlp_recordable
+#   opentelemetry-cpp::otlp_grpc_exporter             - Imported target of opentelemetry-cpp::otlp_grpc_exporter
+#   opentelemetry-cpp::otlp_grpc_log_exporter         - Imported target of opentelemetry-cpp::otlp_grpc_log_exporter
+#   opentelemetry-cpp::otlp_http_client               - Imported target of opentelemetry-cpp::otlp_http_client
+#   opentelemetry-cpp::otlp_http_exporter             - Imported target of opentelemetry-cpp::otlp_http_exporter
+#   opentelemetry-cpp::otlp_http_log_exporter         - Imported target of opentelemetry-cpp::otlp_http_log_exporter
+#   opentelemetry-cpp::ostream_log_exporter           - Imported target of opentelemetry-cpp::ostream_log_exporter
+#   opentelemetry-cpp::ostream_metrics_exporter       - Imported target of opentelemetry-cpp::ostream_metrics_exporter
+#   opentelemetry-cpp::ostream_span_exporter          - Imported target of opentelemetry-cpp::ostream_span_exporter
+#   opentelemetry-cpp::prometheus_exporter_deprecated - Imported target of opentelemetry-cpp::prometheus_exporter_deprecated
+#   opentelemetry-cpp::elasticsearch_log_exporter     - Imported target of opentelemetry-cpp::elasticsearch_log_exporter
+#   opentelemetry-cpp::etw_exporter                   - Imported target of opentelemetry-cpp::etw_exporter
+#   opentelemetry-cpp::jaeger_trace_exporter          - Imported target of opentelemetry-cpp::jaeger_trace_exporter
+#   opentelemetry-cpp::zpages                         - Imported target of opentelemetry-cpp::zpages
+#   opentelemetry-cpp::http_client_curl               - Imported target of opentelemetry-cpp::http_client_curl
 #
 
 # =============================================================================

--- a/exporters/prometheus/CMakeLists.txt
+++ b/exporters/prometheus/CMakeLists.txt
@@ -27,6 +27,12 @@ target_include_directories(
          "$<INSTALL_INTERFACE:include>")
 
 set(PROMETHEUS_EXPORTER_TARGETS prometheus_exporter_deprecated)
+if(TARGET pull)
+  list(APPEND PROMETHEUS_EXPORTER_TARGETS pull)
+endif()
+if(TARGET push)
+  list(APPEND PROMETHEUS_EXPORTER_TARGETS push)
+endif()
 target_link_libraries(
   prometheus_exporter_deprecated
   PUBLIC opentelemetry_metrics_deprecated prometheus-cpp::pull

--- a/exporters/prometheus/CMakeLists.txt
+++ b/exporters/prometheus/CMakeLists.txt
@@ -30,8 +30,8 @@ set(PROMETHEUS_EXPORTER_TARGETS prometheus_exporter_deprecated)
 if(TARGET pull)
   list(APPEND PROMETHEUS_EXPORTER_TARGETS pull)
 endif()
-if(TARGET push)
-  list(APPEND PROMETHEUS_EXPORTER_TARGETS push)
+if(TARGET core)
+  list(APPEND PROMETHEUS_EXPORTER_TARGETS core)
 endif()
 target_link_libraries(
   prometheus_exporter_deprecated

--- a/exporters/prometheus/CMakeLists.txt
+++ b/exporters/prometheus/CMakeLists.txt
@@ -26,7 +26,7 @@ target_include_directories(
   PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>"
          "$<INSTALL_INTERFACE:include>")
 
-set(PROMETHEUS_EXPORTER_TARGETS prometheus_exporter_deprecated pull core)
+set(PROMETHEUS_EXPORTER_TARGETS prometheus_exporter_deprecated)
 target_link_libraries(
   prometheus_exporter_deprecated
   PUBLIC opentelemetry_metrics_deprecated prometheus-cpp::pull

